bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/ICloudService.java

```diff
@@ -246,8 +246,12 @@ public boolean validate2faCode(String code) throws IOException, InterruptedExcep
             this.session.post(AUTH_ENDPOINT + "/verify/trusteddevice/securitycode", JsonUtils.toJson(requestBody),
                     headers);
         } catch (ICloudApiResponseException ex) {
-            logger.debug("Code verification failed.", ex);
-            return false;
+            // iCloud API returns different 4xx error codes even if validation is successful
+            // currently 400 seems to show that verification "really" failed.
+            if (ex.getStatusCode() == 400) {
+                this.logger.debug("Code verification failed.", ex);
+                return false;
```

Comment:
Not a review, but after quickly reading, i think this is too narrow to only look for a 400 error, as it ignores errors in the 50x range, are their specific 40x errors that are known which would probably be preferable to target , or a small range to check for?

Reply:
Currently known is 412 and 421. I was also in between a positive list and this more radical approach. I decided for the later, because in the end ignoring the errors does not really harm and I feared that we will become aware of other errors which should be ignored one-by-one...

Reply:
How about 

`(ex.getStatusCode() == 400 || ex.getStatusCode() >= 500)`

Reply:
@digitaldan :: are you ok with the proposed alternative ?

Reply:
Yeah, i'm good !

Reply:
I have not looked at the code, but do we probably want to make sure we have logging in place so if other 4XX codes break this in the future, its easy to see when DEBUG is enable (or this might be hard to track down if its swallowed)

Reply:
Good idea to also add a DEBUG log showing the status code just before the test.

@maihacke : can you please finalize, with your proposed test and adding a log ?

Reply:
I changed the check as discussed.